crates/sdk/src/masp/utils.rs

```diff
@@ -153,13 +245,426 @@ where
+impl RequestBuilderExt for reqwest::RequestBuilder {
+    #[inline(always)]
+    fn keep_alive(self) -> reqwest::RequestBuilder {
+        self.header("Connection", "Keep-Alive")
+    }
+}
+
+#[cfg(not(target_family = "wasm"))]
+impl IndexerMaspClient {
+    /// Create a new [`IndexerMaspClient`].
+    #[inline]
+    pub fn new(client: reqwest::Client, indexer_api: reqwest::Url) -> Self {
+        let indexer_api = Arc::new(indexer_api);
+        Self {
+            client,
+            indexer_api,
+        }
+    }
+
+    fn endpoint(&self, which: &str) -> String {
+        format!("{}{which}", self.indexer_api)
+    }
+
+    async fn get_server_error(
+        response: reqwest::Response,
+    ) -> Result<String, Error> {
+        use serde::Deserialize;
+
+        #[derive(Deserialize)]
+        struct Response {
+            message: String,
+        }
+
+        let payload: Response = response.json().await.map_err(|err| {
+            Error::Other(format!(
+                "Could not deserialize server's error JSON response: {err}"
+            ))
+        })?;
+
+        Ok(payload.message)
+    }
+}
+
+#[cfg(not(target_family = "wasm"))]
+impl MaspClient for IndexerMaspClient {
+    #[inline(always)]
+    fn capabilities(&self) -> MaspClientCapabilities {
+        MaspClientCapabilities::AllData
+    }
+
+    async fn last_block_height(&self) -> Result<Option<BlockHeight>, Error> {
+        use serde::Deserialize;
+
+        #[derive(Deserialize)]
+        struct Response {
+            block_height: u64,
+        }
+
+        let response = self
+            .client
+            .get(self.endpoint("/height"))
+            .keep_alive()
+            .send()
+            .await
+            .map_err(|err| {
+                Error::Other(format!(
+                    "Failed to fetch latest block height: {err}"
+                ))
+            })?;
+        if !response.status().is_success() {
+            let err = Self::get_server_error(response).await?;
+            return Err(Error::Other(format!(
+                "Failed to fetch last block height: {err}"
+            )));
+        }
+        let payload: Response = response.json().await.map_err(|err| {
+            Error::Other(format!(
+                "Could not deserialize latest block height JSON response: \
+                 {err}"
+            ))
+        })?;
+
+        Ok(if payload.block_height != 0 {
+            Some(BlockHeight(payload.block_height))
+        } else {
+            None
+        })
+    }
+
+    async fn fetch_shielded_transfers<IO: Io>(
+        &self,
+        progress: &impl ProgressTracker<IO>,
+        shutdown_signal: &mut ShutdownSignal,
+        mut tx_sender: FetchQueueSender,
+        BlockHeight(mut from): BlockHeight,
+        BlockHeight(to): BlockHeight,
+    ) -> Result<(), Error> {
+        use serde::Deserialize;
+
+        #[derive(Deserialize)]
+        struct TransactionSlot {
+            // masp_tx_index: u64,
```

Comment:
Should we remove this or do we have plans to use it in the upcoming updates?

Reply:
This will get used in the future